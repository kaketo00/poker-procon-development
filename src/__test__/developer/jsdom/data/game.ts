import type { Game } from '@/schema/game';

export const testGameInfoData: Game = {
  id: '10',
  status: 'progress',
  createdAt: '2024-04-15T18:52:10.296Z',
  players: {
    Player1: {
      name: 'Player1',
      status: 'active',
      point: 37600,
      round: {
        betPoint: 0,
        action: 'bet',
        cards: [
          {
            suit: 'Clubs',
            number: 1,
          },
          {
            suit: 'Diamonds',
            number: 7,
          },
          {
            suit: 'Spades',
            number: 6,
          },
          {
            suit: 'Spades',
            number: 9,
          },
          {
            suit: 'Clubs',
            number: 8,
          },
        ],
        hand: 1,
      },
    },
    Player2: {
      name: 'Player2',
      status: 'active',
      point: 17600,
      round: {
        betPoint: 0,
        action: 'bet',
        cards: [
          {
            suit: 'Hearts',
            number: 13,
          },
          {
            suit: 'Hearts',
            number: 1,
          },
          {
            suit: 'Spades',
            number: 5,
          },
          {
            suit: 'Clubs',
            number: 4,
          },
          {
            suit: 'Hearts',
            number: 4,
          },
        ],
        hand: 2,
      },
    },
    Player3: {
      name: 'Player3',
      status: 'active',
      point: 24800,
      round: {
        betPoint: 0,
        action: 'bet',
        cards: [
          {
            suit: 'Diamonds',
            number: 12,
          },
          {
            suit: 'Diamonds',
            number: 5,
          },
          {
            suit: 'Diamonds',
            number: 2,
          },
          {
            suit: 'Spades',
            number: 1,
          },
          {
            suit: 'Diamonds',
            number: 1,
          },
        ],
        hand: 2,
      },
    },
    Player4: {
      name: 'Player4',
      status: 'out',
      point: 0,
      round: {
        betPoint: 0,
        action: null,
        cards: [],
      },
    },
  },
  totalRound: 100,
  initialPoint: 20000,
  fee: 200,
  seatingOrder: ['Player3', 'Player4', 'Player1', 'Player2'],
  startedAt: '2024-04-15T18:52:13.832Z',
};
