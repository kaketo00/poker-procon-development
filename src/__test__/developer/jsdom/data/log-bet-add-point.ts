import type { Log } from '@/schema/log';

export const testLogBetAddPoint: Log = {
  round: 10,
  player: 'Player1',
  phase: 'bet-1',
  content: { type: 'bet', amount: 314, action: 'bet', payPoint: 314 },
  game: {
    id: '10',
    status: 'progress',
    createdAt: '2024-04-22T12:06:55.706Z',
    players: {
      Player1: {
        name: 'Player1',
        status: 'active',
        point: 64746,
        round: {
          betPoint: 314,
          first: 314,
          second: 0,
          action: 'bet',
          cards: [
            { suit: 'Clubs', number: 6 },
            { suit: 'Spades', number: 8 },
            { suit: 'Hearts', number: 9 },
            { suit: 'Diamonds', number: 1 },
            { suit: 'Spades', number: 1 },
          ],
        },
      },
      Player2: {
        name: 'Player2',
        status: 'active',
        point: 6256,
        round: {
          betPoint: 0,
          first: 0,
          second: 0,
          action: null,
          cards: [
            { suit: 'Diamonds', number: 6 },
            { suit: 'Clubs', number: 7 },
            { suit: 'Diamonds', number: 7 },
            { suit: 'Diamonds', number: 12 },
            { suit: 'Clubs', number: 1 },
          ],
        },
      },
      Player3: {
        name: 'Player3',
        status: 'out',
        point: 0,
        round: { betPoint: 0, first: 0, second: 0, action: null, cards: [] },
      },
      Player4: {
        name: 'Player4',
        status: 'active',
        point: 8084,
        round: {
          betPoint: 0,
          first: 0,
          second: 0,
          action: null,
          cards: [
            { suit: 'Diamonds', number: 3 },
            { suit: 'Hearts', number: 4 },
            { suit: 'Diamonds', number: 4 },
            { suit: 'Spades', number: 11 },
            { suit: 'Hearts', number: 11 },
          ],
        },
      },
    },
    totalRound: 100,
    initialPoint: 20000,
    fee: 200,
    seatingOrder: ['Player2', 'Player1', 'Player4', 'Player3'],
    startedAt: '2024-04-22T12:06:56.923Z',
  },
  pot: 914,
  minBetPoint: 314,
  createdAt: 1713787617161,
};
