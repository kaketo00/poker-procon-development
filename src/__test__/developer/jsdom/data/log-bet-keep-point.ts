import type { Log } from '@/schema/log';

export const testLogBetKeepPoint: Log = {
  round: 10,
  player: 'Player1',
  phase: 'bet-1',
  content: { type: 'bet', amount: -1, action: 'drop', payPoint: 0 },
  game: {
    id: '10',
    status: 'progress',
    createdAt: '2024-04-22T12:06:55.706Z',
    players: {
      Player1: {
        name: 'Player1',
        status: 'active',
        point: 64746,
        round: {
          betPoint: 0,
          first: 0,
          second: 0,
          action: 'drop',
          cards: [],
        },
      },
      Player2: {
        name: 'Player2',
        status: 'active',
        point: 6256,
        round: {
          betPoint: 0,
          first: 0,
          second: 0,
          action: null,
          cards: [
            { suit: 'Diamonds', number: 6 },
            { suit: 'Clubs', number: 7 },
            { suit: 'Diamonds', number: 7 },
            { suit: 'Diamonds', number: 12 },
            { suit: 'Clubs', number: 1 },
          ],
        },
      },
      Player3: {
        name: 'Player3',
        status: 'out',
        point: 0,
        round: { betPoint: 0, first: 0, second: 0, action: null, cards: [] },
      },
      Player4: {
        name: 'Player4',
        status: 'active',
        point: 8084,
        round: {
          betPoint: 0,
          first: 0,
          second: 0,
          action: null,
          cards: [
            { suit: 'Diamonds', number: 3 },
            { suit: 'Hearts', number: 4 },
            { suit: 'Diamonds', number: 4 },
            { suit: 'Spades', number: 11 },
            { suit: 'Hearts', number: 11 },
          ],
        },
      },
    },
    totalRound: 100,
    initialPoint: 20000,
    fee: 200,
    seatingOrder: ['Player2', 'Player1', 'Player4', 'Player3'],
    startedAt: '2024-04-22T12:06:56.923Z',
  },
  pot: 600,
  minBetPoint: 0,
  createdAt: 1713787617161,
};
