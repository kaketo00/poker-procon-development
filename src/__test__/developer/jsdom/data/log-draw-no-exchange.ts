import type { Log } from '@/schema/log';

export const testLogDrawNoExchange: Log = {
  round: 10,
  player: 'Player1',
  phase: 'draw-1',
  content: {
    type: 'draw',
    exchange: [false, false, false, false, false],
    keep: [
      { suit: 'Spades', number: 8 },
      { suit: 'Hearts', number: 9 },
      { suit: 'Diamonds', number: 1 },
      { suit: 'Spades', number: 1 },
      { suit: 'Hearts', number: 1 },
    ],
    discard: [],
    newCard: [],
    before: [
      { suit: 'Spades', number: 8 },
      { suit: 'Hearts', number: 9 },
      { suit: 'Diamonds', number: 1 },
      { suit: 'Spades', number: 1 },
      { suit: 'Hearts', number: 1 },
    ],
    after: [
      { suit: 'Spades', number: 8 },
      { suit: 'Hearts', number: 9 },
      { suit: 'Diamonds', number: 1 },
      { suit: 'Spades', number: 1 },
      { suit: 'Hearts', number: 1 },
    ],
  },
  game: {
    id: '10',
    status: 'progress',
    createdAt: '2024-04-22T12:06:55.706Z',
    players: {
      Player1: {
        name: 'Player1',
        status: 'active',
        point: 64746,
        round: {
          betPoint: 314,
          action: 'bet',
          cards: [
            { suit: 'Spades', number: 8 },
            { suit: 'Hearts', number: 9 },
            { suit: 'Diamonds', number: 1 },
            { suit: 'Spades', number: 1 },
            { suit: 'Hearts', number: 1 },
          ],
        },
      },
      Player2: {
        name: 'Player2',
        status: 'active',
        point: 5942,
        round: {
          betPoint: 314,
          action: 'call',
          cards: [
            { suit: 'Diamonds', number: 6 },
            { suit: 'Clubs', number: 7 },
            { suit: 'Diamonds', number: 7 },
            { suit: 'Diamonds', number: 12 },
            { suit: 'Clubs', number: 1 },
          ],
        },
      },
      Player3: {
        name: 'Player3',
        status: 'out',
        point: 0,
        round: { betPoint: 0, action: null, cards: [] },
      },
      Player4: {
        name: 'Player4',
        status: 'active',
        point: 7770,
        round: {
          betPoint: 314,
          action: 'call',
          cards: [
            { suit: 'Diamonds', number: 3 },
            { suit: 'Hearts', number: 4 },
            { suit: 'Diamonds', number: 4 },
            { suit: 'Spades', number: 11 },
            { suit: 'Hearts', number: 11 },
          ],
        },
      },
    },
    totalRound: 100,
    initialPoint: 20000,
    fee: 200,
    seatingOrder: ['Player2', 'Player1', 'Player4', 'Player3'],
    startedAt: '2024-04-22T12:06:56.923Z',
  },
  pot: 1542,
  minBetPoint: 314,
  createdAt: 1713787617162,
};
