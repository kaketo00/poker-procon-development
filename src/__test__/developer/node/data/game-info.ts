import type { GameInfo } from '@/schema/game';

export const testGameInfoData: GameInfo = {
  currentRound: 1,
  phase: 'start',
  order: ['Player1', 'Player2', 'Player3', 'Player4'],
  pot: 0,
  minBetPoint: 0,
  players: {
    Player1: {
      name: 'Player1',
      status: 'active',
      point: 37600,
      round: {
        betPoint: 0,
        action: 'bet',
        cards: [],
      },
    },
    Player2: {
      name: 'Player2',
      status: 'active',
      point: 17600,
      round: {
        betPoint: 0,
        action: 'bet',
        cards: [],
      },
    },
    Player3: {
      name: 'Player3',
      status: 'active',
      point: 24800,
      round: {
        betPoint: 0,
        action: 'bet',
        cards: [],
      },
    },
    Player4: {
      name: 'Player4',
      status: 'out',
      point: 0,
      round: {
        betPoint: 0,
        action: null,
        cards: [],
      },
    },
  },
  totalRound: 100,
  initialPoint: 20000,
  fee: 200,
};
