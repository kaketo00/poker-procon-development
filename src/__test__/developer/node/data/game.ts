import type { Game } from '@/schema/game';

export const testGameData: Game = {
  id: '1',
  status: 'new',
  createdAt: '2024-05-01T10:15:19.019Z',
  players: {
    DemoPlayer1: {
      name: 'DemoPlayer1',
      status: 'active',
      point: 37600,
      round: {
        betPoint: 0,
        action: 'bet',
        cards: [],
      },
    },
    DemoPlayer2: {
      name: 'DemoPlayer2',
      status: 'active',
      point: 17600,
      round: {
        betPoint: 0,
        action: 'bet',
        cards: [],
      },
    },
    DemoPlayer3: {
      name: 'DemoPlayer3',
      status: 'active',
      point: 24800,
      round: {
        betPoint: 0,
        action: 'bet',
        cards: [],
      },
    },
    DemoPlayer4: {
      name: 'DemoPlayer4',
      status: 'out',
      point: 0,
      round: {
        betPoint: 0,
        action: null,
        cards: [],
      },
    },
  },
  totalRound: 100,
  initialPoint: 20000,
  fee: 200,
  seatingOrder: ['DemoPlayer1', 'DemoPlayer2', 'DemoPlayer3', 'DemoPlayer4'],
  startedAt: '2024-05-01T10:52:16.257Z',
};
