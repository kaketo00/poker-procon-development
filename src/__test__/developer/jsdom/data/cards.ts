import type { Card, CardSet } from '@/schema/card';
import { Hand } from '@/schema/game';

type CardSetKye = 'A' | 'B1' | 'B2';
export const testCards: {
  [hand in Hand]: {
    [key in CardSetKye]: CardSet<Card>;
  };
} = {
  [Hand.RoyalStraightFlush]: {
    A: [
      { suit: 'Spades', number: 10 },
      { suit: 'Spades', number: 11 },
      { suit: 'Spades', number: 12 },
      { suit: 'Spades', number: 13 },
      { suit: 'Spades', number: 1 },
    ], // ロイヤルストレートフラッシュ(10♠ J♠ Q♠ K♠ A♠)
    B1: [
      { suit: 'Clubs', number: 10 },
      { suit: 'Clubs', number: 11 },
      { suit: 'Clubs', number: 12 },
      { suit: 'Clubs', number: 13 },
      { suit: 'Clubs', number: 1 },
    ], // ロイヤルストレートフラッシュ(10♣ J♣ Q♣ K♣ A♣)
    B2: [], // 使用しない
  },
  [Hand.StraightFlush]: {
    A: [
      { suit: 'Spades', number: 9 },
      { suit: 'Spades', number: 10 },
      { suit: 'Spades', number: 11 },
      { suit: 'Spades', number: 12 },
      { suit: 'Spades', number: 13 },
    ], // ストレートフラッシュ(9♠ 10♠ J♠ Q♠ K♠)
    B1: [
      { suit: 'Spades', number: 4 },
      { suit: 'Spades', number: 5 },
      { suit: 'Spades', number: 6 },
      { suit: 'Spades', number: 7 },
      { suit: 'Spades', number: 8 },
    ], // ストレートフラッシュ(4♠ 5♠ 6♠ 7♠ 8♠)
    B2: [
      { suit: 'Diamonds', number: 9 },
      { suit: 'Diamonds', number: 10 },
      { suit: 'Diamonds', number: 11 },
      { suit: 'Diamonds', number: 12 },
      { suit: 'Diamonds', number: 13 },
    ], // ストレートフラッシュ(9♦ 10♦ J♦ Q♦ K♦)
  },
  [Hand.FourOfAKind]: {
    A: [
      { suit: 'Clubs', number: 4 },
      { suit: 'Spades', number: 13 },
      { suit: 'Hearts', number: 13 },
      { suit: 'Diamonds', number: 13 },
      { suit: 'Clubs', number: 13 },
    ], // フォーカード(4♣ K♠ K♥ K♦ K♣)
    B1: [
      { suit: 'Spades', number: 7 },
      { suit: 'Hearts', number: 7 },
      { suit: 'Diamonds', number: 7 },
      { suit: 'Clubs', number: 7 },
      { suit: 'Clubs', number: 12 },
    ], // フォーカード(7♠ 7♦ 7♥ 7♣ Q♣)
    B2: [], // 使用しない
  },
  [Hand.FullHouse]: {
    A: [
      { suit: 'Spades', number: 12 },
      { suit: 'Hearts', number: 12 },
      { suit: 'Spades', number: 13 },
      { suit: 'Hearts', number: 13 },
      { suit: 'Diamonds', number: 13 },
    ], // フルハウス(Q♠ Q♥ K♠ K♠ K♦)
    B1: [
      { suit: 'Spades', number: 2 },
      { suit: 'Hearts', number: 2 },
      { suit: 'Spades', number: 3 },
      { suit: 'Hearts', number: 3 },
      { suit: 'Diamonds', number: 3 },
    ], // フルハウス(2♠ 2♦ 3♦ 3♠ 3♥)
    B2: [], // 使用しない
  },
  [Hand.Flush]: {
    A: [
      { suit: 'Spades', number: 2 },
      { suit: 'Spades', number: 6 },
      { suit: 'Spades', number: 9 },
      { suit: 'Spades', number: 13 },
      { suit: 'Spades', number: 1 },
    ], // フラッシュ(2♠ 6♠ 9♠ K♠ A♠)
    B1: [
      { suit: 'Clubs', number: 3 },
      { suit: 'Clubs', number: 4 },
      { suit: 'Clubs', number: 5 },
      { suit: 'Clubs', number: 7 },
      { suit: 'Clubs', number: 8 },
    ], // フラッシュ(3♠ 4♠ 5♠ 7♠ 8♠)
    B2: [
      { suit: 'Clubs', number: 2 },
      { suit: 'Clubs', number: 6 },
      { suit: 'Clubs', number: 9 },
      { suit: 'Clubs', number: 13 },
      { suit: 'Clubs', number: 1 },
    ], // フラッシュ(2♣ 6♣ 9♣ K♣ A♣)
  },
  [Hand.Straight]: {
    A: [
      { suit: 'Spades', number: 10 },
      { suit: 'Spades', number: 11 },
      { suit: 'Spades', number: 12 },
      { suit: 'Spades', number: 13 },
      { suit: 'Hearts', number: 1 },
    ], // ストーレート(10♠ J♠ Q♠ K♠ A♥)
    B1: [
      { suit: 'Spades', number: 4 },
      { suit: 'Spades', number: 5 },
      { suit: 'Spades', number: 6 },
      { suit: 'Spades', number: 7 },
      { suit: 'Hearts', number: 8 },
    ], // フラッシュ(4♠ 5♦ 6♥ 7♣ 8♦)
    B2: [
      { suit: 'Diamonds', number: 10 },
      { suit: 'Diamonds', number: 11 },
      { suit: 'Diamonds', number: 12 },
      { suit: 'Diamonds', number: 13 },
      { suit: 'Clubs', number: 1 },
    ], // フラッシュ(10♦ J♦ Q♦ K♦ A♣)
  },
  [Hand.ThreeOfAKind]: {
    A: [
      { suit: 'Spades', number: 5 },
      { suit: 'Spades', number: 6 },
      { suit: 'Spades', number: 8 },
      { suit: 'Hearts', number: 8 },
      { suit: 'Diamonds', number: 8 },
    ], // スリーカード(5♠ 6♠ 8♠ 8♥ 8♦)
    B1: [
      { suit: 'Spades', number: 5 },
      { suit: 'Spades', number: 6 },
      { suit: 'Spades', number: 7 },
      { suit: 'Hearts', number: 7 },
      { suit: 'Diamonds', number: 7 },
    ], // スリーカード(3♠ 4♠ 7♠ 7♥ 7♦)
    B2: [], // 使用しない
  },
  [Hand.TwoPair]: {
    A: [
      { suit: 'Spades', number: 3 },
      { suit: 'Spades', number: 5 },
      { suit: 'Hearts', number: 5 },
      { suit: 'Spades', number: 7 },
      { suit: 'Hearts', number: 7 },
    ], // ツーペア(3♠ 5♠ 5♥ 7♠ 7♥)
    B1: [
      { suit: 'Spades', number: 2 },
      { suit: 'Spades', number: 4 },
      { suit: 'Hearts', number: 4 },
      { suit: 'Spades', number: 6 },
      { suit: 'Hearts', number: 6 },
    ], // ツーペア(2♠ 4♠ 4♥ 6♠ 6♥)
    B2: [
      { suit: 'Hearts', number: 3 },
      { suit: 'Diamonds', number: 5 },
      { suit: 'Clubs', number: 5 },
      { suit: 'Diamonds', number: 7 },
      { suit: 'Clubs', number: 7 },
    ], // ツーペア(3♥ 5♦ 5♣ 7♦ 7♣)
  },
  [Hand.OnePair]: {
    A: [
      { suit: 'Spades', number: 6 },
      { suit: 'Hearts', number: 6 },
      { suit: 'Spades', number: 7 },
      { suit: 'Diamonds', number: 8 },
      { suit: 'Clubs', number: 9 },
    ], // ワンペア(6♠ 6♥ 7♠ 8♦ 9♣)
    B1: [
      { suit: 'Spades', number: 4 },
      { suit: 'Hearts', number: 4 },
      { suit: 'Spades', number: 9 },
      { suit: 'Diamonds', number: 10 },
      { suit: 'Clubs', number: 11 },
    ], // ワンペア(4♠ 4♥ 9♥ 10♦ J♣)
    B2: [
      { suit: 'Clubs', number: 6 },
      { suit: 'Diamonds', number: 6 },
      { suit: 'Diamonds', number: 7 },
      { suit: 'Clubs', number: 8 },
      { suit: 'Spades', number: 9 },
    ], // ワンペア(6♣ 6♦ 7♦ 8♣ 9♠)
  },
  [Hand.HighCard]: {
    A: [
      { suit: 'Diamonds', number: 2 },
      { suit: 'Spades', number: 4 },
      { suit: 'Clubs', number: 6 },
      { suit: 'Hearts', number: 7 },
      { suit: 'Spades', number: 1 },
    ], // ハイカード(2♦ 4♠ 6♣ 7♥ A♠)
    B1: [
      { suit: 'Clubs', number: 2 },
      { suit: 'Hearts', number: 4 },
      { suit: 'Diamonds', number: 6 },
      { suit: 'Clubs', number: 7 },
      { suit: 'Spades', number: 13 },
    ], // ハイカード(2♣ 4♥ 6♦ 7♣ K♠)
    B2: [
      { suit: 'Clubs', number: 2 },
      { suit: 'Hearts', number: 4 },
      { suit: 'Diamonds', number: 6 },
      { suit: 'Clubs', number: 7 },
      { suit: 'Hearts', number: 1 },
    ], // ハイカード(2♣ 4♥ 6♦ 7♣ A♥)
  },
  [Hand.Drop]: {
    A: [], // 使用しない
    B1: [], // 使用しない
    B2: [], // 使用しない
  },
};
