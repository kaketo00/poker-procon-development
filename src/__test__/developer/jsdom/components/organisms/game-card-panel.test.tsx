import { ChakraProvider } from '@chakra-ui/react';
import { render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { QueryClient, QueryClientProvider } from 'react-query';

import { testGameInfoData } from '@/__test__/developer/jsdom/data/game';
import type { GameCardPanelProps } from '@/components/organisms/game-card-panel';
import GameCardPanel from '@/components/organisms/game-card-panel';
import { SUCCESS_MESSAGES } from '@/constants/message';
import theme from '@/styles/theme';

jest.mock('@/clients/game', () => ({
  async startGame() {
    return {
      id: '1',
      message: SUCCESS_MESSAGES.startGame,
    };
  },
}));

const Component = (props: GameCardPanelProps) => (
  <ChakraProvider theme={theme}>
    <QueryClientProvider client={new QueryClient()}>
      <GameCardPanel {...props} />
    </QueryClientProvider>
  </ChakraProvider>
);

describe('GameCardPanel', () => {
  test('スナップショット/新規', () => {
    const { asFragment } = render(
      <Component
        game={{ ...testGameInfoData, status: 'new', startedAt: undefined }}
        index={1}
        postStartGame={jest.fn()}
        setIsLoading={jest.fn()}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('スナップショット/進行中', () => {
    const { asFragment } = render(
      <Component
        game={{ ...testGameInfoData, status: 'progress' }}
        index={1}
        postStartGame={jest.fn()}
        setIsLoading={jest.fn()}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('スナップショット/終了', () => {
    const { asFragment } = render(
      <Component
        game={{ ...testGameInfoData, status: 'finished' }}
        index={1}
        postStartGame={jest.fn()}
        setIsLoading={jest.fn()}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('ゲーム開始', async () => {
    const setIsLoading = jest.fn();
    const postStartGame = jest.fn();

    const { getByRole } = render(
      <Component
        game={{ ...testGameInfoData, status: 'new', startedAt: undefined }}
        index={1}
        setIsLoading={setIsLoading}
        postStartGame={postStartGame}
      />
    );

    const button = getByRole('button', { name: '開始' });
    await userEvent.click(button);

    await waitFor(() => {
      expect(setIsLoading).toHaveBeenCalledWith(true);
      expect(postStartGame).toHaveBeenCalled();
    });
  });
});
