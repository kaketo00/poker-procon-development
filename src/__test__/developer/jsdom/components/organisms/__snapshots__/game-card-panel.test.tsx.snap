// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameCardPanel スナップショット/新規 1`] = `
<DocumentFragment>
  <div
    class="css-vxlbt5"
  >
    <div
      class="css-u8yv6t"
    >
      <p
        class="chakra-text css-bq8qtg"
      >
        10
      </p>
      <div
        class="css-14d7ov9"
      >
        <svg
          fill="none"
          height="40"
          viewBox="0 0 40 40"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.5 35.005L34.2046 20.2998C37.9318 16.5725 37.9318 10.5227 34.2046 6.79546C30.4775 3.06818 24.428 3.06818 20.7008 6.79546L19.5 7.99635L18.2992 6.79546C14.572 3.06818 8.5225 3.06818 4.79536 6.79546C1.06821 10.5227 1.06821 16.5725 4.79536 20.2998L19.5 35.005Z"
            fill="#B4171E"
          />
        </svg>
      </div>
    </div>
    <div
      class="css-87dp0f"
    >
      <div
        class="css-patx7f"
      >
        <div
          class="css-1446vam"
        >
          <p
            class="chakra-text css-1d1fz8k"
          >
            ステータス
          </p>
          <span
            class="css-y9e69r"
          >
            新規
          </span>
        </div>
        <div
          class="css-1446vam"
        >
          <p
            class="chakra-text css-1d1fz8k"
          >
            作成日時
          </p>
          <p
            class="chakra-text css-12tg5c1"
          >
            2024-04-16 03:52:10
          </p>
        </div>
        <div
          class="css-1446vam"
        >
          <p
            class="chakra-text css-1d1fz8k"
          >
            開始日時
          </p>
          <p
            class="chakra-text css-12tg5c1"
          />
        </div>
        <div
          class="css-1446vam"
        >
          <p
            class="chakra-text css-1d1fz8k"
          >
            プレイヤー
          </p>
          <div
            class="css-1yyugsn"
          >
            <p
              class="chakra-text css-npyw9q"
            >
              Player1
            </p>
            <p
              class="chakra-text css-npyw9q"
            >
              Player2
            </p>
            <p
              class="chakra-text css-npyw9q"
            >
              Player3
            </p>
            <p
              class="chakra-text css-npyw9q"
            >
              Player4
            </p>
          </div>
        </div>
        <div
          class="css-1446vam"
        >
          <p
            class="chakra-text css-1d1fz8k"
          >
            ラウンド数
          </p>
          <p
            class="chakra-text css-12tg5c1"
          >
            100round
          </p>
        </div>
      </div>
      <button
        class="chakra-button css-a29uer"
        type="button"
      >
        開始
      </button>
    </div>
    <div
      class="css-ocrfjm"
    >
      <p
        class="chakra-text css-bq8qtg"
      >
        10
      </p>
      <div
        class="css-14d7ov9"
      >
        <svg
          fill="none"
          height="40"
          viewBox="0 0 40 40"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.5 35.005L34.2046 20.2998C37.9318 16.5725 37.9318 10.5227 34.2046 6.79546C30.4775 3.06818 24.428 3.06818 20.7008 6.79546L19.5 7.99635L18.2992 6.79546C14.572 3.06818 8.5225 3.06818 4.79536 6.79546C1.06821 10.5227 1.06821 16.5725 4.79536 20.2998L19.5 35.005Z"
            fill="#B4171E"
          />
        </svg>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`GameCardPanel スナップショット/終了 1`] = `
<DocumentFragment>
  <div
    class="css-vxlbt5"
  >
    <div
      class="css-u8yv6t"
    >
      <p
        class="chakra-text css-bq8qtg"
      >
        10
      </p>
      <div
        class="css-14d7ov9"
      >
        <svg
          fill="none"
          height="40"
          viewBox="0 0 40 40"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.5 35.005L34.2046 20.2998C37.9318 16.5725 37.9318 10.5227 34.2046 6.79546C30.4775 3.06818 24.428 3.06818 20.7008 6.79546L19.5 7.99635L18.2992 6.79546C14.572 3.06818 8.5225 3.06818 4.79536 6.79546C1.06821 10.5227 1.06821 16.5725 4.79536 20.2998L19.5 35.005Z"
            fill="#B4171E"
          />
        </svg>
      </div>
    </div>
    <div
      class="css-87dp0f"
    >
      <div
        class="css-patx7f"
      >
        <div
          class="css-1446vam"
        >
          <p
            class="chakra-text css-1d1fz8k"
          >
            ステータス
          </p>
          <span
            class="css-b5iegr"
          >
            終了
          </span>
        </div>
        <div
          class="css-1446vam"
        >
          <p
            class="chakra-text css-1d1fz8k"
          >
            作成日時
          </p>
          <p
            class="chakra-text css-12tg5c1"
          >
            2024-04-16 03:52:10
          </p>
        </div>
        <div
          class="css-1446vam"
        >
          <p
            class="chakra-text css-1d1fz8k"
          >
            開始日時
          </p>
          <p
            class="chakra-text css-12tg5c1"
          >
            2024-04-16 03:52:13
          </p>
        </div>
        <div
          class="css-1446vam"
        >
          <p
            class="chakra-text css-1d1fz8k"
          >
            プレイヤー
          </p>
          <div
            class="css-1yyugsn"
          >
            <p
              class="chakra-text css-npyw9q"
            >
              Player1
            </p>
            <p
              class="chakra-text css-npyw9q"
            >
              Player2
            </p>
            <p
              class="chakra-text css-npyw9q"
            >
              Player3
            </p>
            <p
              class="chakra-text css-npyw9q"
            >
              Player4
            </p>
          </div>
        </div>
        <div
          class="css-1446vam"
        >
          <p
            class="chakra-text css-1d1fz8k"
          >
            ラウンド数
          </p>
          <p
            class="chakra-text css-12tg5c1"
          >
            100round
          </p>
        </div>
      </div>
      <a
        class="chakra-button css-jtpj8e"
        href="/games/10"
      >
        詳細
      </a>
    </div>
    <div
      class="css-ocrfjm"
    >
      <p
        class="chakra-text css-bq8qtg"
      >
        10
      </p>
      <div
        class="css-14d7ov9"
      >
        <svg
          fill="none"
          height="40"
          viewBox="0 0 40 40"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.5 35.005L34.2046 20.2998C37.9318 16.5725 37.9318 10.5227 34.2046 6.79546C30.4775 3.06818 24.428 3.06818 20.7008 6.79546L19.5 7.99635L18.2992 6.79546C14.572 3.06818 8.5225 3.06818 4.79536 6.79546C1.06821 10.5227 1.06821 16.5725 4.79536 20.2998L19.5 35.005Z"
            fill="#B4171E"
          />
        </svg>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`GameCardPanel スナップショット/進行中 1`] = `
<DocumentFragment>
  <div
    class="css-vxlbt5"
  >
    <div
      class="css-u8yv6t"
    >
      <p
        class="chakra-text css-bq8qtg"
      >
        10
      </p>
      <div
        class="css-14d7ov9"
      >
        <svg
          fill="none"
          height="40"
          viewBox="0 0 40 40"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.5 35.005L34.2046 20.2998C37.9318 16.5725 37.9318 10.5227 34.2046 6.79546C30.4775 3.06818 24.428 3.06818 20.7008 6.79546L19.5 7.99635L18.2992 6.79546C14.572 3.06818 8.5225 3.06818 4.79536 6.79546C1.06821 10.5227 1.06821 16.5725 4.79536 20.2998L19.5 35.005Z"
            fill="#B4171E"
          />
        </svg>
      </div>
    </div>
    <div
      class="css-87dp0f"
    >
      <div
        class="css-patx7f"
      >
        <div
          class="css-1446vam"
        >
          <p
            class="chakra-text css-1d1fz8k"
          >
            ステータス
          </p>
          <span
            class="css-1v5fpkt"
          >
            進行中
          </span>
        </div>
        <div
          class="css-1446vam"
        >
          <p
            class="chakra-text css-1d1fz8k"
          >
            作成日時
          </p>
          <p
            class="chakra-text css-12tg5c1"
          >
            2024-04-16 03:52:10
          </p>
        </div>
        <div
          class="css-1446vam"
        >
          <p
            class="chakra-text css-1d1fz8k"
          >
            開始日時
          </p>
          <p
            class="chakra-text css-12tg5c1"
          >
            2024-04-16 03:52:13
          </p>
        </div>
        <div
          class="css-1446vam"
        >
          <p
            class="chakra-text css-1d1fz8k"
          >
            プレイヤー
          </p>
          <div
            class="css-1yyugsn"
          >
            <p
              class="chakra-text css-npyw9q"
            >
              Player1
            </p>
            <p
              class="chakra-text css-npyw9q"
            >
              Player2
            </p>
            <p
              class="chakra-text css-npyw9q"
            >
              Player3
            </p>
            <p
              class="chakra-text css-npyw9q"
            >
              Player4
            </p>
          </div>
        </div>
        <div
          class="css-1446vam"
        >
          <p
            class="chakra-text css-1d1fz8k"
          >
            ラウンド数
          </p>
          <p
            class="chakra-text css-12tg5c1"
          >
            100round
          </p>
        </div>
      </div>
      <a
        class="chakra-button css-t46x3c"
        disabled=""
        href=""
      >
        進行中
      </a>
    </div>
    <div
      class="css-ocrfjm"
    >
      <p
        class="chakra-text css-bq8qtg"
      >
        10
      </p>
      <div
        class="css-14d7ov9"
      >
        <svg
          fill="none"
          height="40"
          viewBox="0 0 40 40"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.5 35.005L34.2046 20.2998C37.9318 16.5725 37.9318 10.5227 34.2046 6.79546C30.4775 3.06818 24.428 3.06818 20.7008 6.79546L19.5 7.99635L18.2992 6.79546C14.572 3.06818 8.5225 3.06818 4.79536 6.79546C1.06821 10.5227 1.06821 16.5725 4.79536 20.2998L19.5 35.005Z"
            fill="#B4171E"
          />
        </svg>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
